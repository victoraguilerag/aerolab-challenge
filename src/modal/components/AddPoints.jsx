import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import Value from './Value'

export function AddPoints (props) {
  const { handleAddPoints, handleValueSelected, selectedValue } = props
  const values = [ 1000, 5000, 7500 ]

  return (
    <div className="add-points" style={styles.addPoints}>
      <div>
        <div className="add-points-text" style={styles.addPointsText}>
          Select points
        </div>
        <div className="values" style={styles.values}>
          {
            values.map(value => (
              <Value
                key={value}
                active={(value === selectedValue)}
                value={value}
                handleValueSelected={handleValueSelected}
              />
            ))
          }
        </div>
      </div>
      <div className="reedem-buttom" style={styles.reedemButton} onClick={handleAddPoints}>
        <p className="reedem-text" style={styles.reedemText}>
          Add points
        </p>
      </div>
    </div>
  )
}

const styles = {
  'addPoints': {
    'height': '70vh',
    'width': '50%',
    'display': 'flex',
    'flexFlow': 'column',
    'justifyContent': 'center',
    'alignItems': 'center',
    'borderLeft': 'thin solid lightgray'
  },
  'addPointsText': {
    'fontSize': '24px',
    'color': '#616161',
    'letterSpacing': '-0.15px',
    'lineHeight': '48px',
    'textAlign': 'center',
    'margin': '1em auto'
  },
  'values': {
    'listStyle': 'none',
    'width': '50%',
    'height': '45px',
    'margin': '0 auto',
    'display': 'flex',
    'flexFlow': 'row',
    'justifyContent': 'center',
    'padding': '0'
  },
  'reedemButton': {
    'backgroundImage': 'linear-gradient(-180deg, #0ad4fa 0%, #25bbf1 100%)',
    'width': '228px',
    'height': '42px',
    'margin': '40px auto',
    'borderRadius': '5px',
    'display': 'flex',
    'justifyContent': 'center',
    'alignItems': 'center',
    'cursor': 'pointer'
  },
  'reedemText': {
    'fontSize': '18px',
    'color': 'white',
    'letterSpacing': '-0.04px',
    'textAlign': 'center'
  }
}

AddPoints.propTypes = {
  handleAddPoints: PropTypes.func.isRequired,
  selectedValue: PropTypes.number.isRequired,
  handleValueSelected: PropTypes.func.isRequired
}

const mapStateToProps = (state) => {
  return {
    selectedValue: state.selectedValue
  }
}

export default connect(mapStateToProps)(AddPoints)
