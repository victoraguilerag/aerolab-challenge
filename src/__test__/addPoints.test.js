import React from 'react'
import { AddPoints } from '../modal/components/AddPoints.jsx'
import renderer from 'react-test-renderer'

const functionMock = () => console.log('function mock');

test('AddPoints renders', () => {
  const component = renderer.create(
    <AddPoints
      handleAddPoints={functionMock}
      handleValueSelected={functionMock}
      selectedValue={1000}
    />,
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
