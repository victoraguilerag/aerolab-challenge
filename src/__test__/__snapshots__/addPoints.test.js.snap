// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddPoints renders 1`] = `
<div
  className="add-points"
  style={
    Object {
      "alignItems": "center",
      "borderLeft": "thin solid lightgray",
      "display": "flex",
      "flexFlow": "column",
      "height": "70vh",
      "justifyContent": "center",
      "width": "50%",
    }
  }
>
  <div>
    <div
      className="add-points-text"
      style={
        Object {
          "color": "#616161",
          "fontSize": "24px",
          "letterSpacing": "-0.15px",
          "lineHeight": "48px",
          "margin": "1em auto",
          "textAlign": "center",
        }
      }
    >
      Select points
    </div>
    <div
      className="values"
      style={
        Object {
          "display": "flex",
          "flexFlow": "row",
          "height": "45px",
          "justifyContent": "center",
          "listStyle": "none",
          "margin": "0 auto",
          "padding": "0",
          "width": "50%",
        }
      }
    >
      <div
        className="active-value"
        style={
          Object {
            "background": "#0ad4fa",
            "borderRadius": "100px",
            "cursor": "pointer",
            "height": "40px",
            "margin": "0 5px",
          }
        }
      >
        <div
          className="active-value-text"
          onClick={[Function]}
          style={
            Object {
              "alignItems": "center",
              "color": "#ffffff",
              "display": "flex",
              "fontSize": "1em",
              "height": "100%",
              "justifyContent": "center",
              "margin": "0 10px",
              "textAlign": "center",
            }
          }
        >
          1000
          <img
            alt="coin"
            className="coin"
            height="24px"
            src={Object {}}
            style={
              Object {
                "margin": "0 3px",
              }
            }
            width="24px"
          />
        </div>
      </div>
      <div
        className="value"
        onClick={[Function]}
        style={
          Object {
            "background": "#ededed",
            "borderRadius": "100px",
            "cursor": "pointer",
            "height": "40px",
            "margin": "0 10px",
          }
        }
      >
        <div
          className="valueText"
          style={
            Object {
              "alignItems": "center",
              "color": "#a3a3a3",
              "display": "flex",
              "fontSize": "1em",
              "height": "100%",
              "justifyContent": "center",
              "margin": "0 10px",
              "textAlign": "center",
            }
          }
        >
          5000
          <img
            alt="coin"
            className="coin"
            height="24px"
            src={Object {}}
            style={
              Object {
                "margin": "0 3px",
              }
            }
            width="24px"
          />
        </div>
      </div>
      <div
        className="value"
        onClick={[Function]}
        style={
          Object {
            "background": "#ededed",
            "borderRadius": "100px",
            "cursor": "pointer",
            "height": "40px",
            "margin": "0 10px",
          }
        }
      >
        <div
          className="valueText"
          style={
            Object {
              "alignItems": "center",
              "color": "#a3a3a3",
              "display": "flex",
              "fontSize": "1em",
              "height": "100%",
              "justifyContent": "center",
              "margin": "0 10px",
              "textAlign": "center",
            }
          }
        >
          7500
          <img
            alt="coin"
            className="coin"
            height="24px"
            src={Object {}}
            style={
              Object {
                "margin": "0 3px",
              }
            }
            width="24px"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="reedem-buttom"
    onClick={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundImage": "linear-gradient(-180deg, #0ad4fa 0%, #25bbf1 100%)",
        "borderRadius": "5px",
        "cursor": "pointer",
        "display": "flex",
        "height": "42px",
        "justifyContent": "center",
        "margin": "40px auto",
        "width": "228px",
      }
    }
  >
    <p
      className="reedem-text"
      style={
        Object {
          "color": "white",
          "fontSize": "18px",
          "letterSpacing": "-0.04px",
          "textAlign": "center",
        }
      }
    >
      Add points
    </p>
  </div>
</div>
`;
